File: Hackerrank/30 Days of Code/Day26_Nested_Logic.py

str_input = input()
a_day, a_mon, a_year = str_input.split()
str_input = input()
e_day, e_mon, e_year = str_input.split()

a_day, a_mon, a_year= int(a_day), int(a_mon), int(a_year)
e_day, e_mon, e_year= int(e_day), int(e_mon), int(e_year)
fine=0

if((a_year == e_year) and (a_mon == e_mon) and (a_day<=e_day)):
    fine = 0
elif((a_year == e_year) and (a_mon == e_mon) and (a_day > e_day)):
    fine = (a_day - e_day) * 15
elif((a_year == e_year) and (a_mon > e_mon)):
    fine = (a_mon - e_mon) * 500
elif(a_year > e_year):
        fine = 10000
        
print(fine)

File: Hackerrank/30 Days of Code/Day26_Nested_Logic.c
#include <stdio.h>
#include <string.h>
#include <math.h>
#include <stdlib.h>

int main() {

    int a_day, a_mon, a_year;
    int e_day, e_mon, e_year;
    int fine=0;
    scanf("%d %d %d", &a_day, &a_mon, &a_year);
    scanf("%d %d %d", &e_day, &e_mon, &e_year);

    if((a_year == e_year) && (a_mon == e_mon) && (a_day<=e_day))
        fine = 0;
    else if((a_year == e_year) && (a_mon == e_mon) && (a_day > e_day))
        fine = (a_day - e_day) * 15;
    else if((a_year == e_year) && (a_mon > e_mon))
        fine = (a_mon - e_mon) * 500;
    else if(a_year > e_year)
        fine = 10000;
        
    printf("%d", fine);
        
    return 0;
}
